presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java

```diff
@@ -51,7 +53,9 @@ public ISO8601HashRedisValueProvider(DecoderColumnHandle columnHandle, String va
         @Override
         public long getLong()
         {
-            long millis = FORMATTER.parseMillis(getSlice().toStringAscii());
+            TemporalAccessor temporal = FORMATTER.parseBest(getSlice().toStringAscii(), LocalDateTime::from, LocalDate::from);
+            LocalDateTime dateTime = temporal instanceof LocalDateTime ? LocalDateTime.from(temporal) : LocalDate.from(temporal).atStartOfDay();
+            long millis = dateTime.toInstant(ZoneOffset.UTC).toEpochMilli();
```

Comment:
can you add some test cases for testing the different combinations of strings with date, datetime etc

Reply:
added a few test cases. lmk if this looks good @ajaygeorge